src/api.py

```diff
@@ -67,7 +73,13 @@ def _search_posts_query(username: str, search_content: str) -> Optional[dict]:
                 },
             ],
         }
+    if not username:  # avoid an empty $or clause which will cause an error
```

Comment:
Sorry let me give a specific example of what I mean:
```python
def _posts_by_user_query(username: str) -> Optional[dict]:
    username_query = {}
    if not username:  # avoid an empty $or clause which will cause an error
        return None

    formatted_username = _normalize_username(username)
    username_query = {
        "$or": [
            {
                "username": formatted_username,
            },
            {
                "comments.username": formatted_username,
            },
            {
                "echo.username": formatted_username,
            },
        ],
    }

    return username_query
```

The `if` occurs at the beginning of the function and immediately returns if the condition does not pass. That prevents any extra unnecessary processing and moves some stuff out of an indent.

Reply:
Looks good to me!